server/fleet/errors.go

```diff
@@ -279,6 +279,36 @@ func (e PermissionError) PermissionError() []map[string]string {
 	return forbidden
 }
 
+// OTAForbiddenError is a special kind of forbidden error that intentionally
+// exposes information about the error so it can be shown in iPad/iPhone native
+// dialogs during OTA enrollment.
+//
+// I couldn't find any documentation but the way it works is:
+//
+// - if the response has a status code 403
+// - and the body has a `message` field
+//
+// the content of `message` will be displayed to the end user.
```

Comment:
Looks kind of similar to the software update required [error](https://github.com/fleetdm/fleet/blob/8012a055edb6ac335ba557412cb6acee1fb03a8b/server/fleet/apple_mdm.go#L913-L914). I wonder if the docs you're looking for are somewhere in the same vicinity of the Apple MDM repo.

Reply:
ohhh thanks, I will take a look